src/gas/fetchFeeHistory.ts

```diff
@@ -0,0 +1,170 @@
+import { BN } from 'ethereumjs-util';
+import { stripHexPrefix } from 'ethjs-util';
+import { query } from '../util';
+
+type HexString = `0x${string}`;
+
+/**
+ * @type FeeHistory
+ *
+ * Historical data for gas fees for a range of blocks within a particular network.
+ * @property startBlockId - The id of the oldest block in the block range requested.
+ * @property blocks - Data for the blocks in the block range requested, sorted from earliest to
+ * latest.
+ */
+export type FeeHistory<Percentile extends number> = {
+  startBlockId: HexString;
+  blocks: BlockFeeHistory<Percentile>[];
+};
+
+/**
+ * @type BlockFeeHistory
+ *
+ * Historical data for gas fees for a particular block.
+ * @property baseFeePerGas - The base fee per gas for the block.
+ * @property gasUsedRatio - A number between 0 and 1 that represents the gas paid for the block vs.
+ * its set gas limit.
+ * @property priorityFeesByPercentile - The priority fees paid for the transactions in the block
+ * that occurred at particular levels at which those transactions contributed to the overall gas
+ * used for the block, indexed by those percentiles. (See docs for `fetchFeeHistory` for more on how
+ * this works.)
+ */
+type BlockFeeHistory<Percentile extends number> = {
+  baseFeePerGas: number;
+  gasUsedRatio: number;
+  priorityFeesByPercentile: Record<Percentile, number>;
+};
+
+/**
+ * @type EthFeeHistoryResponse
+ *
+ * Response data for `eth_feeHistory`.
+ * @property oldestBlock - The id of the oldest block (in hex format) in the range of blocks
+ * requested.
+ * @property baseFeePerGas - Base fee per gas for each block in the range of blocks requested.
+ * @property gasUsedRatio - A number between 0 and 1 that represents the gas used vs. gas limit for
+ * each block in the range of blocks requested.
+ * @property reward - The priority fee at the percentiles requested for each block in the range of
+ * blocks requested.
+ */
+
+export type EthFeeHistoryResponse = {
+  oldestBlock: HexString;
+  baseFeePerGas: HexString[];
+  gasUsedRatio: number[];
+  reward: HexString[][];
+};
+
+/**
+ * Converts a hexadecimal string to a decimal number.
+ *
+ * @param hex - A string encoding a hexadecimal number (with a "0x" prefix).
+ * @returns The number in decimal.
+ */
+function hexToDec(hex: string): number {
+  return parseInt(new BN(stripHexPrefix(hex), 16).toString(10), 10);
+}
+
+/**
+ * Converts a decimal number to a hexadecimal string.
+ *
+ * @param dec - A number in decimal.
+ * @returns A string encoding that number in hexadecimal (with a "0x" prefix).
+ */
+function decToHex(dec: number): string {
+  const hexString = new BN(dec.toString(), 10).toString(16);
```

Comment:
Is there a way we can do this as well without using BN?

Reply:
Update: This is badly named since this doesn't take a decimal number but just an integer. Also this could take `string` in addition to `number`.